creator/ingest_runs/models.py

```diff
@@ -73,6 +82,35 @@ class Meta:
         help_text="The user who submitted this ingest run",
     )
 
+    @transition(field=state, source="waiting", target="running")
+    def start(self):
+        """
+        Begin running the ingest process.
+        """
+        return
+
+    @transition(field=state, source="running", target="complete")
+    def complete(self):
+        """
+        Finish running the ingest process without error.
+        """
+        return
+
+    @transition(field=state, source="*", target="canceled")
```

Comment:
Hmm not sure about this but I think should only be able to cancel the IngestRun if its in running state because if its not in running state then there's nothing to cancel (the redis job would not even be created at this point)